rerun_cpp/src/rerun/recording_stream.hpp

```diff
@@ -216,36 +220,44 @@ namespace rerun {
         }
 
         template <typename C, typename... Ts>
-        static void push_data_cells(
+        static Status push_data_cells(
             std::vector<DataCell>& data_cells, const std::vector<C>& first, const Ts&... rest
         ) {
-            // TODO(andreas): Error handling.
-            const auto cell = C::to_data_cell(first.data(), first.size()).ValueOrDie();
-            data_cells.push_back(cell);
-            push_data_cells(data_cells, rest...);
+            const auto cell_result = C::to_data_cell(first.data(), first.size());
+            if (!cell_result.ok()) {
```

Comment:
we can follow the Rust name convention? 🤷 

```suggestion
            if (cell_result.is_err()) {
```